relay-server/src/actors/global_config.rs

```diff
@@ -0,0 +1,242 @@
+use std::borrow::Cow;
+use std::sync::Arc;
+
+use relay_config::Config;
+use relay_dynamic_config::GlobalConfig;
+use relay_system::{Addr, AsyncResponse, FromMessage, Interface, Service};
+use reqwest::Method;
+use serde::{Deserialize, Serialize};
+use tokio::sync::{mpsc, watch};
+
+use crate::actors::upstream::{
+    RequestPriority, SendQuery, UpstreamQuery, UpstreamRelay, UpstreamRequestError,
+};
+use crate::utils::SleepHandle;
+
+/// The result of sending a global config query to upstream. It can fail both in sending it,
+/// and in the response.
+type UpstreamQueryResult =
+    Result<Result<GetGlobalConfigResponse, UpstreamRequestError>, relay_system::SendError>;
+
+/// The response of a fetch of a global config from upstream.
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GetGlobalConfigResponse {
+    #[serde(default)]
+    global: Option<GlobalConfig>,
+}
+
+/// The request to fetch a global config from upstream.
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GetGlobalConfig {
+    pub global: bool,
+    // Dummy variable - upstream expects a list of public keys.
+    public_keys: Vec<()>,
+}
+
+impl GetGlobalConfig {
+    pub fn query() -> GetGlobalConfig {
+        GetGlobalConfig {
+            global: true,
+            public_keys: vec![],
+        }
+    }
+}
+
+impl UpstreamQuery for GetGlobalConfig {
+    type Response = GetGlobalConfigResponse;
+
+    fn method(&self) -> reqwest::Method {
+        Method::POST
+    }
+
+    fn path(&self) -> std::borrow::Cow<'static, str> {
+        Cow::Borrowed("/api/0/relays/projectconfigs/?version=4")
+    }
+
+    fn retry() -> bool {
+        false
+    }
+
+    fn priority() -> super::upstream::RequestPriority {
+        RequestPriority::High
+    }
+
+    fn route(&self) -> &'static str {
+        "global_config"
+    }
+}
+
+/// The message for requesting the most recent global config from [`GlobalConfigService`].
+pub struct Get;
+
+/// The message for receiving a watch that subscribes to the [`GlobalConfigService`].
+pub struct Subscribe;
+
+/// A way to get updates of the global config.
```

Comment:
It would be great to add more docs about `GlobalConfigManager` here, describe what it does and why and what this interface represents.